apps/aestratum/src/aestratum_config.erl

```diff
@@ -0,0 +1,195 @@
+    SessionCfg1 = SessionCfg#{desired_solve_time => DesiredSolveTime * 1000,
+                              max_solve_time => MaxSolveTime * 1000},
+    maps:merge(maps:without([session], Result), SessionCfg1);
+configure(reward, #{reward := #{beneficiaries := PoolShareBins,
+                                reward_last_rounds := LastN,
+                                keys := [{<<"dir">>, KeysDir}]}} = Result) ->
+    (is_integer(LastN) andalso LastN > 0 andalso LastN < 10)
+        orelse error({invalid, reward_last_rounds}),
+
+    ContractAddr = case aec_governance:get_network_id() of
+                       <<"ae_mainnet">> -> ?PAYMENT_CONTRACT_MAINNET_ADDRESS;
+                       _Other -> ?PAYMENT_CONTRACT_TESTNET_ADDRESS
+                   end,
+    ContractPK   = aestratum_conv:contract_address_to_pubkey(ContractAddr),
+
+    {CallerPK, CallerSK} = CallerKeyPair = read_keys(KeysDir, await),
+    CallerAddr   = aestratum_conv:account_pubkey_to_address(CallerPK),
+    check_keypair_roundtrips(CallerKeyPair) orelse error(invalid_keypair),
+
+    PoolPercentShares =
+        lists:foldl(fun (Bnf, Acc) ->
+                            [<<"ak_", _/binary>> = AccountAddr, PctShareBin] =
+                                binary:split(Bnf, <<":">>),
+                            PctShare = aestratum_conv:binary_to_number(PctShareBin),
+                            Acc#{AccountAddr => maps:get(AccountAddr, Acc, 0) + PctShare}
+                    end, #{}, PoolShareBins),
+    PoolPercentSum = lists:sum(maps:values(PoolPercentShares)),
+    PoolPercentSum < 100.0 orelse error(beneficiaries_reward_too_high),
+
+    maps:merge(maps:without([reward], Result),
+               #{last_n                => LastN,
+                 contract_pubkey       => ContractPK,
+                 contract_address      => ContractAddr,
+                 caller_pubkey         => CallerPK,
+                 caller_privkey        => CallerSK,
+                 caller_address        => CallerAddr,
+                 pool_percent_sum      => PoolPercentSum,
+                 pool_percent_shares   => PoolPercentShares,
+                 reward_keyblock_delay => ?DEFAULT_REWARD_KEYBLOCK_DELAY,
+                 payout_keyblock_delay => ?DEFAULT_PAYOUT_KEYBLOCK_DELAY}).
+
+
+check_extra_nonce_bytes(MaxConnections, NumAcceptors, ExtraNonceBytes) ->
+    ((MaxConnections + NumAcceptors) * 2) < aestratum_nonce:max(ExtraNonceBytes).
+
+check_keypair_roundtrips({PK, SK}) ->
+    Sig = enacl:sign_detached(<<"roundtrip">>, SK),
+    is_ok(enacl:sign_verify_detached(Sig, <<"roundtrip">>, PK)).
+
+read_keys(Dir) ->
+    AbsDir = case filename:pathtype(Dir) of
+                 relative -> filename:join(code:priv_dir(aestratum), Dir);
+                 absolute -> Dir
+             end,
+    PKPath = filename:join(AbsDir, <<"sign_key.pub">>),
+    SKPath = filename:join(AbsDir, <<"sign_key">>),
```

Comment:
No attempt to hide the content here, I guess that is ok in a trusted environment?

Reply:
No reason I think. This account isn't accumulating the rewards - all what comes in is already scheduled to go out. 

Maybe, after 10 years of mining, there could be some letftovers accumulated but it would be only tiny fractions of mining rewards.